src/test/java/com/google/devtools/build/lib/analysis/ToolchainResolverTest.java

```diff
@@ -433,6 +437,102 @@ public void resolve_noMatchingPlatform() throws Exception {
         .isInstanceOf(NoMatchingPlatformException.class);
   }
 
+  @Test
+  public void unloadedToolchainContext_load() throws Exception {
+    addToolchain(
+        "extra",
+        "extra_toolchain_linux",
+        ImmutableList.of("//constraints:linux"),
+        ImmutableList.of("//constraints:linux"),
+        "baz");
+    rewriteWorkspace(
+        "register_toolchains('//extra:extra_toolchain_linux')",
+        "register_execution_platforms('//platforms:linux')");
+
+    useConfiguration("--platforms=//platforms:linux");
+    ResolveToolchainsKey key =
+        ResolveToolchainsKey.create("test", ImmutableSet.of(testToolchainType), targetConfigKey);
+
+    // Create the UnloadedToolchainContext.
+    EvaluationResult<ResolveToolchainsValue> result = createToolchainContextBuilder(key);
+    assertThatEvaluationResult(result).hasNoError();
+    UnloadedToolchainContext unloadedToolchainContext = result.get(key).unloadedToolchainContext();
+    assertThat(unloadedToolchainContext).isNotNull();
+
+    // Create the prerequisites.
+    ConfiguredTargetAndData toolchain = getConfiguredTargetAndData(
+        Label.parseAbsoluteUnchecked("//extra:extra_toolchain_linux_impl"), targetConfig);
+    OrderedSetMultimap<Attribute, ConfiguredTargetAndData> prerequisiteMap = OrderedSetMultimap.create();
+    prerequisiteMap.put(Attribute.attr(PlatformSemantics.RESOLVED_TOOLCHAINS_ATTR, LABEL_LIST).build(), toolchain);
+
+    ToolchainContext toolchainContext = unloadedToolchainContext.load(prerequisiteMap);
+    assertThat(toolchainContext).isNotNull();
+    assertThat(toolchainContext.forToolchainType(testToolchainType)).isNotNull();
+    assertThat(toolchainContext.forToolchainType(testToolchainType).hasField("data")).isTrue();
+    assertThat(toolchainContext.forToolchainType(testToolchainType).getValue("data")).isEqualTo("baz");
+  }
+
+  @Test
+  public void unloadedToolchainContext_load_withTemplateVariables() throws Exception {
```

Comment:
What's the significance of "unloaded" in this naming?

Reply:
UnloadedToolchainContext is the class that the ToolchainResolver generates, and the load() method on UnloadedToolchainContext is what is being tested.